tsl/src/compression/api.c

```diff
@@ -812,7 +813,15 @@ tsl_create_compressed_chunk(PG_FUNCTION_ARGS)
 										  numrows_pre_compression,
 										  numrows_post_compression);
 
+	chunk_was_compressed = ts_chunk_is_compressed(cxt.srcht_chunk);
 	ts_chunk_set_compressed_chunk(cxt.srcht_chunk, compress_ht_chunk->fd.id);
+	if (!chunk_was_compressed) {
+		/* The chunk was not compressed, before it had the compressed chunk
+		 * attached to it, so it must set it to be partial.
+		 * TODO: actually, we should check if the chunk contains uncompressed rows
```

Comment:
The way that this is implemented, it over-eagerly marks chunks as partial. What would the right way be to determine if the chunk has uncompressed rows in it?

Reply:
I've fixed this.